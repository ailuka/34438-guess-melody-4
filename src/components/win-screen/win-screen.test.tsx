import * as React from "react";
import * as renderer from "react-test-renderer";
import WinScreen from "./win-screen";
import {Router} from "react-router-dom";
import history from "../../history";

describe(`WinScreen`, () => {
  describe(`should render correctly with 3 questions`, () => {
    it(`with 0 mistakes`, () => {
      const winScreen = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={3}
              mistakesCount={0}
              onReplayButtonClick={() => null}
            />
          </Router>
      ).toJSON();

      expect(winScreen).toMatchSnapshot();
    });

    it(`with 2 mistakes`, () => {
      const winScreen = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={3}
              mistakesCount={2}
              onReplayButtonClick={() => null}
            />
          </Router>
      ).toJSON();

      expect(winScreen).toMatchSnapshot();
    });
  });

  describe(`should render correctly with 2 questions`, () => {
    it(`with 0 mistakes`, () => {
      const winScreen = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={2}
              mistakesCount={0}
              onReplayButtonClick={() => null}
            />
          </Router>
      ).toJSON();

      expect(winScreen).toMatchSnapshot();
    });

    it(`with 1 mistake`, () => {
      const winScreen = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={2}
              mistakesCount={1}
              onReplayButtonClick={() => null}
            />
          </Router>
      ).toJSON();

      expect(winScreen).toMatchSnapshot();
    });
  });
});
