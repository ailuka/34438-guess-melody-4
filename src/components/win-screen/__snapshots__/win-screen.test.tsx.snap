// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinScreen should render correctly with 2 questions with 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    2
     вопросов и совершили 
    0
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen should render correctly with 2 questions with 1 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    1
     вопросов и совершили 
    1
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen should render correctly with 3 questions with 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    3
     вопросов и совершили 
    0
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen should render correctly with 3 questions with 2 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    1
     вопросов и совершили 
    2
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;
